Jenkinsfile

```diff
@@ -0,0 +1,131 @@
+}
+
+def runTests(name, suffix) {
+  try {
+    sh 'mkdir -p test-output'
+    sh 'chmod 777 test-output'
+    sh "docker-compose -p $name-$suffix -f docker-compose.test.yaml run $name"
+
+  } finally {
+    sh "docker-compose -p $name-$suffix -f docker-compose.test.yaml down -v"
+    junit 'test-output/junit.xml'
+    // clean up files created by node/ubuntu user that cannot be deleted by jenkins. Note: uses global environment variable
+    sh "docker run -u node --mount type=bind,source=$WORKSPACE/test-output,target=/usr/src/app/test-output $name rm -rf test-output/*"
+  }
+}
+
+def pushContainerImage(registry, credentialsId, imageName, tag) {
+  docker.withRegistry("https://$registry", credentialsId) {
+    sh "docker-compose build --no-cache"
+    sh "docker tag $imageName $registry/$imageName:$tag"
+    sh "docker push $registry/$imageName:$tag"
+  }
+}
+
+def deployPR(credentialsId, registry, imageName, tag, extraCommands) {
+  withKubeConfig([credentialsId: credentialsId]) {
+    def deploymentName = "$imageName-$tag"
+    sh "kubectl get namespaces $deploymentName || kubectl create namespace $deploymentName"
+    sh "helm upgrade $deploymentName --install --namespace $deploymentName --atomic ./helm/$imageName --set image=$registry/$imageName:$tag $extraCommands"
+  }
+}
+
+def undeployPR(credentialsId, imageName, tag) {
+  withKubeConfig([credentialsId: credentialsId]) {
+    def deploymentName = "$imageName-$tag"
+    sh "helm delete --purge $deploymentName || echo error removing deployment $deploymentName"
+    sh "kubectl delete $deploymentName"
+  }
+}
+
+def publishChart(imageName) {
+  // jenkins doesn't tidy up folder, remove old charts before running
+  sh "rm -rf helm-charts"
+  sshagent(credentials: ['helm-chart-creds']) {
+    sh "git clone git@gitlab.ffc.aws-int.defra.cloud:helm/helm-charts.git"
+    dir('helm-charts') {
+      sh 'helm init -c'
+      sh "helm package ../helm/$imageName"
+      sh 'helm repo index .'
+      sh 'git config --global user.email "buildserver@defra.gov.uk"'
+      sh 'git config --global user.name "buildserver"'
+      sh 'git checkout master'
+      sh 'git add -A'
+      sh "git commit -m 'update $imageName helm chart from build job'"
+      sh 'git push'
+    }
+  }
+}
+
+node {
+  checkout scm
+  stage('Set branch, PR, and containerTag variables') {
+    (branch, pr, containerTag, mergedPrNo) = getVariables(repoName)
+  }
+  stage('Build test image') {
+    buildTestImage(testImageName, BUILD_NUMBER)
+  }
+  stage('Run tests') {
+      runTests(testImageName, BUILD_NUMBER)
+  }
+  stage('Build production image') {
+    buildProductionImage(imageName, BUILD_NUMBER)
+  }
+  stage('Push container image') {
+    pushContainerImage(registry, regCredsId, imageName, containerTag)
+  }
+  if (pr != '') {
+    stage('Helm install') {
+      def extraCommands = "--values ./helm/$repoName/jenkins-aws.yaml --set name=ffc-demo-api-gateway-$containerTag"
```

Comment:
There's no need to set the name Helm value here, it's set for the web as the ingress needs it but it can just stay as the default in the values file here.

Reply:
Good point, I've removed that.